proposals/dark_mode_extension_icons.md

```diff
@@ -166,10 +166,10 @@ effectively have a wild-card for `color_schemes`. For example, `["*"]` is valid.
 **Misc**
 1. If the top-level `icon_variants` key is provided, the top level `icons` key
 will be ignored.
-1. `icon_variants` will not cause an error in the event that it is invalid.
+1. `icon_variants` will not cause an error if it's invalid.
 Instead, only the faulty icon group(s) will be ignored, with an optional
 warning. Warnings are preferred over errors because they're more adaptable to
-changes in the future.
+changes in the future. If none of the icon groups are valid, should this error?
```

Comment:
```suggestion
changes in the future. If none of the icon groups are valid, agents are expected to fallback to the `icons` property in `manifest.json`. If `icons` is missing, agents are expected to use a default error and not give a hard error.
```

This behaviour matches the current Safari implementation and reliefs extension authors from having to declare additional fallbacks in `icon_variants` for browsers which do not support SVG for example. We may want to revisit this in MV4+ when the `icons` property might be deprecated/discontinued.

Reply:
Safari _does not_ fallback to `icons`. We have separate fallback to the containing app's icon if we can't get the icon from `icon_variants`.

Reply:
@xeenon thanks for clarifying. You are right. In my tests, invalid `manifest.action.icon_variants` would fallback to `manifest.icons` if invalid. Which led me to the false conclusion this also happens for `manifest.icon_variants`. In that case this is up for discussion once again. For browsers not supporting SVG icons falling back to manifest.icons would be great for webcompat. However doing so would make the case for `icon_variants` as a replacement for `icons` weaker.

Reply:
Incorporated change but made it more relaxed such that falling back to icons could be done, but isn't suggested. It's a middle of the road approach vs either requiring it or forbidding it.